test/unit/strands/strands_transpiler.js

```diff
@@ -0,0 +1,56 @@
+import { detectOutsideVariableReferences } from '../../../src/strands/strands_transpiler.js';
+import { suite, test } from '../../../test/js/spec.js';
+
+suite('Strands Transpiler - Outside Variable Detection', function() {
+  test('should allow outer scope variables in uniform callbacks', function() {
+    // OK: mouseX in uniform callback is allowed
+    const code = `
+      baseMaterialShader.modify(() => {
+        const myUniform = uniformFloat(() => mouseX);
+        getWorldPosition((inputs) => {
+          inputs.position.x += myUniform;
+          return inputs;
+        });
+      });
+    `;
+    
+    const errors = detectOutsideVariableReferences(code);
+    assert.equal(errors.length, 0, 'Should not error - mouseX is OK in uniform callback');
+  });
+
+  test('should detect undeclared variable in strand code', function() {
+    // ERROR: mouseX in strand code is not declared
+    const code = `
+      baseMaterialShader.modify(() => {
+        getWorldPosition((inputs) => {
+          inputs.position.x += mouseX; // mouseX not declared in strand!
+          return inputs;
+        });
+      });
+    `;
+    
+    const errors = detectOutsideVariableReferences(code);
+    assert.ok(errors.length > 0, 'Should detect error');
+    assert.ok(errors.some(e => e.variable === 'mouseX'), 'Should detect mouseX');
+  });
+
+  test('should not error when variable is declared', function() {
+    const code = `
+      baseMaterialShader.modify(() => {
```

Comment:
I think this is testing something different from what the function would actually receive. This looks like code that would be found in a p5 sketch, but just the contents of the `modify` callback is sent into the your function normally.

I think a good next step would be to try to do a full integration test by running `yarn dev:global` and editing the sketch in `preview/global`. If that works, you can also show the maintainers by building the library with `npm run build`, and then uploading `lib/p5.js` to a p5 web editor sketch so that others can run the same test. I suspect doing that will reveal some of the problems with the implementation more quickly than in review (but feel free to reach out if you run into issues while doing so!)